frontend/gateway/client/client.go

```diff
@@ -17,7 +18,7 @@ type Client interface {
 type Reference interface {
 	ReadFile(ctx context.Context, req ReadRequest) ([]byte, error)
 	// StatFile(ctx context.Context, req StatRequest) (*StatResponse, error)
-	// ReadDir(ctx context.Context, req ReadDirRequest) ([]*StatResponse, error)
+	ReadDir(ctx context.Context, req ReadDirRequest) ([]*fsutil.Stat, error)
```

Comment:
Need to figure out how to reduce the dependency here as `fsutil` itself has too many for all frontends to include(eg. even a package from moby). Probably just move the type definitions into a subpackage. This can be a follow up to this PR. Or redefine only stat in here and convert in the `ReadDir` handler.

Reply:
Not too keen on duplicating the types and converting since they can get out of sync all too easily.

BTW I was just hacking on fsutil to make it easier to support `StatFile` here and I found that `go test ./` failed with digest mismatches. Not sure what's going on there.

Reply:
> Not sure what's going on there.

Wrong go (1.10 not 1.11) version (not sure why it mattered, but it apparently did).

Reply:
I refactored the type in https://github.com/tonistiigi/fsutil/pull/52, can either wait for that and rebase or do a followup as you prefer.